### PR TITLE
Feature/bug151052 Order custom fields by priority

### PR DESCRIPTION
**Pull request by dmitryTsarevich**

### CODE REVIEW

File: src/manifest.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "Name": "Custom Field Constraints",
     "ShortDescription": "This mashup allows a custom field to be required when an entity is moved to a specific state or a specific value is selected in another custom field.",
     "CompatibleTpVersion": {
-            "Minimum": "3.6.6"
+            "Minimum": "3.8.8"
```
Comment (dmitryTsarevich) [inline src/manifest.json]: `CustomField.NumericPriority` was added in this version.

File: src/shared/services/interrupt/quickAdd.js [hunk 2]
```diff
@@ -70,12 +71,20 @@ const getCustomFieldTemplate = (customField, types) => {
...
+const findNewCustomFieldIndex = (items, newNumericPriority) => {
 
-    const relationsItemIndex = findLastIndex(items, (item) =>
-        item.id === 'SlaveRelations:RelationType' || item.id === 'MasterRelations:RelationType');
+    const nextItemIndex = findIndex(items, (item) => item.numericPriority > newNumericPriority);
 
-    return relationsItemIndex === -1 ? items.length : relationsItemIndex;
+    if (nextItemIndex === -1) {
+
+        const relationsItemIndex = findLastIndex(items, (item) =>
```
Comment (dmitryTsarevich) [inline src/shared/services/interrupt/quickAdd.js]: `Relations` dropdown should be always last, custom fields should come before.

File: src/shared/services/axes.js [hunk 3]
```diff
@@ -39,11 +39,13 @@ const matchTeamWorkflowEntityStateFlat = (valueToMatch, entityState) =>
 
 const matchEntityStateFlat = (valueToMatch, entityType, entityState) => {
 
+    if (!equalIgnoreCase(entityState.entityType.name, entityType.name)) return false;
```
Comment (victor-homyakov) [inline src/shared/services/axes.js]: Style: always use brackets

File: src/shared/services/axes.js [hunk 4]
```diff
@@ -39,11 +39,13 @@ const matchTeamWorkflowEntityStateFlat = (valueToMatch, entityState) =>
...
+
     const isTeamWorkflowEntityState = isObject(valueToMatch);
 
     return isTeamWorkflowEntityState ?
         matchTeamWorkflowEntityStateFlat(valueToMatch, entityState) :
-        equalIgnoreCase(valueToMatch, entityState.name) || equalByShortcut(valueToMatch, entityState);
+            equalIgnoreCase(valueToMatch, entityState.name) || equalByShortcut(valueToMatch, entityState);
```
Comment (victor-homyakov) [inline src/shared/services/axes.js]: Style: no extra indentation needed here

File: src/shared/services/interrupt/quickAdd.js [hunk 5]
```diff
@@ -70,12 +71,20 @@ const getCustomFieldTemplate = (customField, types) => {
...
 
-const findNewCustomFieldIndex = (items) => {
+const findNewCustomFieldIndex = (items, newNumericPriority) => {
 
-    const relationsItemIndex = findLastIndex(items, (item) =>
-        item.id === 'SlaveRelations:RelationType' || item.id === 'MasterRelations:RelationType');
+    const nextItemIndex = findIndex(items, (item) => item.numericPriority > newNumericPriority);
 
-    return relationsItemIndex === -1 ? items.length : relationsItemIndex;
+    if (nextItemIndex === -1) {
```
Comment (victor-homyakov) [inline src/shared/services/interrupt/quickAdd.js]: I'd prefer

```js
if (nextItemIndex >= 0) {
    return nextItemIndex;
}
...
```
to reduce indentation and to move last usage of `nextItemIndex` closer to its definition.

**Pull request merged**